extensions/v1/knative/spec.yaml

```diff
@@ -0,0 +1,40 @@
+applicationCreator: knative.dev
+applicationMaintainer: knative.dev
+applicationProvider: knative.dev
+documentationUrl: https://knative.dev/docs/
+logoUrl: https://avatars2.githubusercontent.com/u/35583233?s=200&v=4
+longDescription: |
+  Knative (pronounced kay-nay-tiv) extends Kubernetes to provide a set of middleware components that are essential to build modern, source-centric, and container-based applications that can run anywhere: on premises, in the cloud, or even in a third-party data center.
+
+  Each of the components under the Knative project attempt to identify common patterns and codify the best practices that are shared by successful, real-world, Kubernetes-based frameworks and applications. Knative components focus on solving mundane but difficult tasks such as:
+
+  * Deploying a container
+  * Routing and managing traffic with blue/green deployment
+  * Scaling automatically and sizing workloads based on demand
+  * Binding running services to eventing ecosystems
+  * Developers on Knative can use familiar idioms, languages, and frameworks to deploy functions, applications, or containers workloads.
+name: knative
+repositoryUrl: github.com/knative/serving
+shortDescription: |
+  Kubernetes-based platform to build, deploy, and manage modern serverless workloads.
+versions:
+- manifestsArchive:
+    uri: https://storage.googleapis.com/sm-marketplace-registry/serving-0.8.0.tar.gz
+  flavors:
+  - name: istio
+    description: "Install Knative using Istio as the Knative Gateway"
+    requirementSets:
+    - meshRequirement:
+        meshType: ISTIO
+        versions:
+          minVersion: "1.1.0"
+  version: "0.8.0-istio"
+- manifestsArchive:
+    uri: https://storage.googleapis.com/sm-marketplace-registry/serving-core-0.8.0.tar.gz
+  flavors:
+  - name: gloo
```

Comment:
The absence of mesh requirements is interpreted as "compatible with no meshes", so this flavor won't be installable. Until we support meshless extensions, our workaround for this is to add istio and linkerd meshRequirements with very low minVersions.